agent/acs/handler/refresh_credentials_handler.go

```diff
@@ -160,6 +167,15 @@ func (refreshHandler *refreshCredentialsHandler) handleSingleMessage(message *ec
 		}
 		if roleType == credentials.ExecutionRoleType {
 			task.SetExecutionRoleCredentialsID(aws.StringValue(message.RoleCredentials.CredentialsId))
+			// Refresh domainless gMSA plugin credentials
+			if task.RequiresDomainlessCredentialSpecResource() {
+				err = checkAndSetDomainlessGMSATaskExecutionRoleCredentialsImpl(iamRoleCredentials, task.Arn, task.RequiresDomainlessCredentialSpecResource())
```

Comment:
I think we should pass `task` to this function and let it extract the ARN and figure out if it's a domainless gMSA task. Passing the result of `task.RequiresDomainlessCredentialSpecResource()` still leaks implementation details into the ACS refresh handler.